src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -404,10 +407,21 @@ public boolean isChatColorsEnabled() {
     }
 
     @Override
+    public void sendThroughMessage(Text message) {
+        checkNotNull(message, "The message text cannot be null!");
+        this.connection.sendPacket(new SPacketChat(SpongeTexts.toComponent(message)));
+    }
+
+    @Override
     public void sendMessage(ChatType type, Text message) {
         checkNotNull(type, "type");
         checkNotNull(message, "message");
 
+        if (this.conversation != null && this.conversation.getChatHandler(this).isPresent()
+            && !this.conversation.getChatHandler(this).get().process(message)) {
+            return;
+        }
+
         ITextComponent component = SpongeTexts.toComponent(message);
```

Comment:
So I don't want this logic within `sendThroughMessage` actually, but I would like some input.

The `sendThroughMessage` method was created, because you can set a conversation to catch `Conversant` input, which would include the messages you are trying to send to them while asking questions and similar. I couldn't think of the greatest way to accomplish besides making a new method which bypasses this catching. 

I would be happy to have some input on this.